File: app/schemas/refund.py
# File: app/schemas/refund.py

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import RefundStatus


class RefundCreateRequest(BaseModel):
    """بيانات طلب استرداد جديد لطلب موجود."""

    refund_amount: Decimal = Field(gt=0)
    currency_code: str | None = None
    reason: str | None = None


class RefundDecisionRequest(BaseModel):
    """ملاحظة اختيارية ترافق قرار رفض طلب استرداد."""

    notes: str | None = None


class RefundOut(BaseModel):
    """تمثيل طلب استرداد كامل في الاستجابات."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    order_id: int
    refund_amount: Decimal
    currency_code: str | None
    reason: str | None
    status: RefundStatus
    processed_by: int | None
    processed_at: datetime | None

File: frontend/src/types/refund.ts
// File: frontend/src/types/refund.ts

import type { RefundStatus } from "@/types/enums";

/** مطابق لـ app.schemas.refund.RefundOut. */
export interface RefundOut {
  id: number;
  order_id: number;
  order_number: string;
  refund_amount: string;
  currency_code: string | null;
  reason: string | null;
  status: RefundStatus;
  processed_by: number | null;
  processed_at: string | null;
}

/** مطابق لـ app.schemas.refund.RefundCreateRequest. */
export interface RefundCreateRequest {
  refund_amount: string;
  currency_code: string | null;
  reason: string | null;
}

/** مطابق لـ app.schemas.refund.RefundDecisionRequest. */
export interface RefundDecisionRequest {
  notes: string | null;
}
